ntoskrnl/po/ntapi.c

```diff
@@ -0,0 +1,981 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Power Manager NT API system calls
+ * COPYRIGHT:   Copyright 2023 George Bișoc <george.bisoc@reactos.org>
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS ********************************************************************/
+
+WORK_QUEUE_ITEM PopUnlockMemoryWorkItem;
+KEVENT PopUnlockMemoryCompleteEvent;
+
+/* PRIVATE FUNCTIONS **********************************************************/
+
+_Use_decl_annotations_
+VOID
+NTAPI
+PopUnlockMemoryWorker(
+    _In_ PVOID Parameter)
+{
+    UNIMPLEMENTED;
+    return;
+}
+
+static
+BOOLEAN
+PopIsCallerPrivileged(
+    _In_ POWER_INFORMATION_LEVEL Level,
+    _In_ PVOID InputBuffer,
+    _In_ KPROCESSOR_MODE PreviousMode)
+{
+    LUID Privilege;
+
+    PAGED_CODE();
+
+    /* If the caller was coming from the kernel, he has absolute privileges */
+    if (PreviousMode == KernelMode)
+    {
+        return TRUE;
+    }
+
+    /*
+     * This is coming from UM of which we cannot trust it, and we might have an
+     * input buffer. Ensure that the caller has the appropriate privilege for
+     * whatever operation he wants to do.
+     */
+    if (InputBuffer)
```

Comment:
The `VerifySystemPolicyAc/Dc` and `VerifyProcessorPowerPolicyAc/Dc` may not require any privilege check, even though an InputBuffer is passed, because these information levels  are asking the Po to just do verification checks on the policy buffer passed in input, but are not doing any actual power action on the system.

Reply:
I was adding this comment in order to not forget to mention an observation about something, that I noticed during testing on windows, and I was checking whether this was already done in your PR.